File: ruby/matrix/matrix.rb
# frozen_string_literal: true

class Matrix
  attr_accessor :rows

  def initialize(string_matrix)
    @rows = string_to_rows(string_matrix)
  end

  def columns
    @columns ||= rows.transpose
  end

  private

  def string_to_rows(string_matrix)
    string_matrix.lines.map { |row_string| row_string.split.map(&:to_i) }
  end
end

File: typescript/matrix/matrix.ts
export class Matrix {
  readonly rows: number[][]
  readonly columns: number[][]

  constructor(matrixStr: string) {
    this.rows = matrixStr
      .split("\n")
      .map((rowStr) => rowStr.split(" ").map(Number))

    this.columns = this
      .rows[0]
      .map((_, rowIndex) => this.rows.map((row) => row[rowIndex]));
  }
}
